internal/server/pod_scheduling_policy.go

```diff
@@ -0,0 +1,96 @@
+// everest
+// Copyright (C) 2025 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+
+	everestv1alpha1 "github.com/percona/everest-operator/api/v1alpha1"
+	"github.com/percona/everest/api"
+	"github.com/percona/everest/pkg/common"
+)
+
+// ListPodSchedulingPolicy lists all pod scheduling policies.
+func (e *EverestServer) ListPodSchedulingPolicy(ctx echo.Context, params api.ListPodSchedulingPolicyParams) error {
+	list, err := e.handler.ListPodSchedulingPolicies(ctx.Request().Context(), &params)
+	if err != nil {
+		e.l.Errorf("ListPodSchedulingPolicies failed: %v", err)
+		return err
+	}
+	return ctx.JSON(http.StatusOK, list)
+}
+
+// CreatePodSchedulingPolicy creates a new pod scheduling policy.
+func (e *EverestServer) CreatePodSchedulingPolicy(c echo.Context) error {
+	psp := &everestv1alpha1.PodSchedulingPolicy{}
+	if err := e.getBodyFromContext(c, psp); err != nil {
+		return errors.Join(errFailedToReadRequestBody, err)
+	}
+
+	if psp.GetNamespace() == "" {
+		psp.Namespace = common.SystemNamespace
+	}
```

Comment:
this means that if in the user request the namespace is other than `common.SystemNamespace` the request will use the original namespace, let's say it's `my-namespace`. But later in the `CreatePodSchedulingPolicy` handlers implementation we have hardcoded `common.SystemNamespace` as the namespace. Looks like we need to throw an error if the user provided a namespace other than `common.SystemNamespace`. 

Same for `UpdatePodSchedulingPolicy`